akka-cluster-metrics/src/main/scala/akka/cluster/metrics/ClusterMetricsExtension.scala

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.cluster.metrics
+
+import akka.actor.ExtendedActorSystem
+import akka.actor.Extension
+import akka.actor.SupervisorStrategy
+import akka.event.LoggingAdapter
+import akka.event.Logging
+import com.typesafe.config.Config
+import scala.collection.immutable
+import akka.actor.Props
+import akka.actor.Deploy
+import akka.actor.ExtensionId
+import akka.actor.ExtensionIdProvider
+import akka.actor.ActorSystem
+import akka.actor.ActorRef
+
+/**
+ * Cluster metrics extension.
+ *
+ * Cluster metrics is primarily for load-balancing of nodes. It controls metrics sampling
+ * at a regular frequency, prepares highly variable data for further analysis by other entities,
+ * and publishes the latest cluster metrics data around the node ring and local eventStream
+ * to assist in determining the need to redirect traffic to the least-loaded nodes.
+ *
+ * Metrics sampling is delegated to the [[MetricsCollector]].
+ *
+ * Smoothing of the data for each monitored process is delegated to the
+ * [[EWMA]] for exponential weighted moving average.
+ */
+class ClusterMetricsExtension(system: ExtendedActorSystem) extends Extension {
+
+  private val log: LoggingAdapter = Logging(system, getClass.getName)
```

Comment:
this is only used at one place, for a failure condition
create it inline at that place instead to avoid unnecessary work/allocations for the normal case